#!/usr/bin/env python
PACKAGE = "tuw_iwos_odometer"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("publish_odom_message", bool_t, 0, "publish odom message", True)
gen.add("broadcast_odom_transform", bool_t, 0, "broadcast odom transform", True)

gen.add("revolute_velocity_tolerance", double_t, 0, "tolerance for revolute velocity matching",  0.025, 0.0, 1.0)
gen.add("steering_position_tolerance", double_t, 0, "tolerance for steering position matching", 0.025, 0.0, 1.0)
gen.add("calculation_iterations", int_t, 0, "number of odometer calculation iterations", 100, 0, 1000000)

gen.add("revolute_noise", double_t, 0, "noise for revolute measurements",  0.025, 0.0, 10.0)
gen.add("steering_noise", double_t, 0, "noise for steering measurements",  0.025, 0.0, 10.0)

exit(gen.generate(PACKAGE, "tuw_iwos_odometer", "MixedOdometer"))
